src/core/server/http/http_server.test.ts

```diff
@@ -1077,4 +1078,37 @@ describe('setup contract', () => {
       expect(isTlsEnabled).toBe(false);
     });
   });
+
+  describe('#getServerInfo', () => {
+    it('returns correct information', async () => {
+      let { getServerInfo } = await server.setup(config);
+
+      expect(getServerInfo()).toEqual({
+        host: '127.0.0.1',
+        name: 'kibana',
+        port: 10002,
+        protocol: 'http',
+      });
+
+      ({ getServerInfo } = await server.setup({
+        ...config,
+        port: 12345,
+        name: 'custom-name',
+        host: 'localhost',
+      }));
+
+      expect(getServerInfo()).toEqual({
+        host: 'localhost',
+        name: 'custom-name',
+        port: 12345,
+        protocol: 'http',
+      });
+    });
+
+    it('returns correct protocol when ssl is enabled', async () => {
+      const { getServerInfo } = await server.setup(configWithSSL);
+
+      expect(getServerInfo().protocol).toEqual('https');
+    });
+  });
```

Comment:
Test for when `protocol` is `socket`?

Reply:
I will be honest, I have no idea how to configure the server to get a socket protocol (only copied the possible values from the HAPI's `config.protocol`)